misk/src/main/kotlin/misk/hibernate/Id.kt

```diff
@@ -0,0 +1,8 @@
+package misk.hibernate
+
+import java.io.Serializable
+
+/** Type-safe persistent identifier, mapped to a long column. */
+data class Id<T : DbEntity<T>>(val id: Long) : Serializable {
```

Comment:
do you think `Long` is too restrictive?

Reply:
I think it's the right type for DB-internal row IDs. It's big enough to be unique and small enough to be efficient within MySQL.